tests/jax/test_fused_attn.py

```diff
@@ -322,7 +322,7 @@ def set_input(self, b, s_q, s_kv, h, d, dtype, attn_mask_type, pad_ratio):
                                         axis=-1)
         self.scaling_factor = 1. / math.sqrt(d)
         self.dropout_probability = 0.
-        self.dropout_rng = jax.random.PRNGKey(0)
+        self.dropout_rng = jax.random.PRNGKey(0) if self.dropout_probability > 0 else None
```

Comment:
I can add some dropout sanity test, but the dropout correctness is hard to implement in a unittest since we can't have the bitmatch results.

Reply:
From the description, this PR change the current behavior when fmha is used vs when it is not used.
This isn't good. We need to do something about this.
Could we make both version behave the same?
If not, I think it will need to be an option.

Reply:
`is_training` and `dropout` sanity check are added in f3d21a1e183aee17889ec87dbd8d85854e407847

Reply:
> Could we make both version behave the same?
If not, I think it will need to be an option.

No, we can't make it behave the same. The cuDNN dropout doesn't support the [dropout shape](https://github.com/NVIDIA/TransformerEngine/blob/v0.8.0.1/transformer_engine/jax/flax/transformer.py#L167). I can add an option for it.

Reply:
Why can't we use the cudnn dropout mask shape as the default one?
That way w/wo fmha we will always have the same dropout mask.
The only diff with be the rng used.

Reply:
I changed the default dropout mask shape to be the same as the cudnn's in 4669452f56bab88723e3a8da14a6f427d73509e2. The PR description is also revised, please take a look. Thanks